packages/rum-core/src/performance-monitoring/metrics.js

```diff
@@ -269,3 +373,21 @@ export class PerfEntryRecorder {
     this.po.disconnect()
   }
 }
+
+export function bootstrapMetrics() {
+  metrics.wasHidden = document.visibilityState === 'hidden'
+
+  window.addEventListener(
+    'visibilitychange',
+    () => {
+      if (document.visibilityState === 'hidden') {
+        metrics.wasHidden = true
+      }
+    },
+    { capture: true, once: true }
+  )
+
+  if (!isPerfEntryTypeSupported(FIRST_INPUT)) {
+    polyfillFID()
```

Comment:
I think this polyfill wont work for libraries and works only for users who can include it on the HTML head document.